### ISSUE TITLE
golden_code appears unset in CodeEnvBatch

### ISSUE THREAD

**Issue by LaoKuiZe** [pettingllms-ai/PettingLLMs#35]
Hi PettingLLMs team,

Thanks for releasing the code. I am trying to reproduce the CodeContests multi-agent training setup, and I noticed a possible issue in the code environment reward path.

In the current `main` branch, the CodeContests loading path seems to preserve a reference solution:

- `scripts/dataprocess/load_code.py` builds rows with a `"solution"` field for CodeContests.
- `pettingllms/multi_agent_env/code/code_utils.py::format_problem()` also returns `"solution"`.

`CodeEnvState` also defines a `golden_code` field, and `UnitTestGenerationAgent.step()` only evaluates generated tests against the golden/reference code when `golden_code` is present:

```python
golden_code = getattr(env_data.state, "golden_code", None)

if gen_inputs and gen_outputs and golden_code:
    passed_ratio, passed_cases, failed_cases = await evaluate_code_against_tests(
        env_data.state.golden_code, gen_inputs, gen_outputs, ...
    )
    env_data.state.generated_test_vs_golden_code_match_ratio = passed_ratio
```

However, in `CodeEnvBatch.__init__`, the state appears to be initialized without passing the loaded `solution` into `golden_code`:

```python
state = CodeEnvState(
    problem=problem["question"],
    ground_truth_test_input=ground_truth_test_input,
    ground_truth_test_output=ground_truth_test_output,
)
```

As a result, `env_data.state.golden_code` seems to remain `None`, so the `generated_test_vs_golden_code_match_ratio` branch is skipped. Since the tester reward is currently:

```python
self.agent_reward = (
    env_data.state.generated_test_vs_golden_code_match_ratio
    + env_data.state.ground_truth_test_vs_generated_code_match_ratio
)
```

the first term may always stay at its default value `0`, and the tester reward effectively becomes only the coder's ground-truth-test pass ratio.

Could you confirm whether this is intended? Thanks!
